### ISSUE TITLE
Split up the code that calculates data for component plots

### ISSUE THREAD

**Issue by tcuongd** [facebook/prophet#1988]
There have been a couple of issues asking how to extract the underlying data for the component plots, e.g. https://github.com/facebook/prophet/issues/1920

I think it wouldn't hurt us to separate the code that calculates the data for the plots from the plotting code.

**Comment by pranith7867** [comment]
@tcuongd Hello can i work on this issue i am looking for a good first issue to work

**Comment by DEEKSHACodeaim** [comment]
@tcuongd I would love to work on this issue under hacktober 2023, please let me know if this is still a requirement ASAP

**Comment by tcuongd** [comment]
Yep I think it'd be a good task still. Pulling out the logic into private methods to calculate the data for the trend plot, seasonality plots, and extra regressor plots would be great. Methods parametrized by the name of the component would be good too.

**Comment by ImZackAdams** [comment]
Is this still being requested or is this a dead issue?
